compiler/rustc_session/src/session.rs

```diff
@@ -284,6 +290,8 @@ impl Session {
     }
 
     #[rustc_lint_diagnostics]
+    #[allow(rustc::untranslatable_diagnostic)]
+    #[allow(rustc::diagnostic_outside_of_impl)]
```

Comment:
I'll take a look at fixing the lint for this case today.

Reply:
#101230

Reply:
Thanks for the PR, David :)

Reply:
@davidtwco The lint still throws an error with functions annotated with `#[rustc_lint_diagnostics]` in `compiler/rustc_session/src/session.rs` :(

Do you want that I take a look at this?

Reply:
That would be helpful!

Reply:
Alright, I'll try to take a look at this today :)

Reply:
I didn't have the time to check this issue :(

Will try to do it on Sunday  🌤️

Reply:
Hey David, I finally had the time to take a look at this and it is working just fine. 

It wasn't work before because my rust buld was messed up and a `./x.py clean` solved the issue.

I only removed the `allow`s from the functions that were annotated with `#[rustc_lint_diagnostics]`.